components/script/dom/intersectionobserver.rs

```diff
@@ -271,6 +302,352 @@ impl IntersectionObserver {
+        time: CrossProcessInstant,
+        root_bounds: Rect<Au>,
+        can_gc: CanGc,
+    ) {
+        for target in &*self.observation_targets.borrow() {
+            // Step 1
+            // > Let registration be the IntersectionObserverRegistration record in target’s internal
+            // > [[RegisteredIntersectionObservers]] slot whose observer property is equal to observer.
+            let registration = target.get_intersection_observer_registration(self).unwrap();
+
+            // Step 2
+            // > If (time - registration.lastUpdateTime < observer.delay), skip further processing for target.
+            if time - registration.last_update_time.get() <
+                Duration::from_millis(self.delay.get().max(0) as u64)
+            {
+                return;
+            }
+
+            // Step 3
+            // > Set registration.lastUpdateTime to time.
+            registration.last_update_time.set(time);
+
+            // Step 4
+            // Let:
+            // > - thresholdIndex be 0.
+            // > - isIntersecting be false.
+            // > - targetRect be a DOMRectReadOnly with x, y, width, and height set to 0.
+            // > - intersectionRect be a DOMRectReadOnly with x, y, width, and height set to 0.
+            let mut threshold_index = 0;
+            let mut is_intersecting = false;
+            let mut target_rect: Rect<Au> = Rect::zero();
+            let mut intersection_rect: Rect<Au> = Rect::zero();
+
+            // If some conditions that requires us to skips "processing further", we will skips those steps.
+            // Current draft specs skipped wrong steps, as it should skip computing fields that would
+            // result in different intersection entry other than the default entry per published spec.
+            // <https://www.w3.org/TR/intersection-observer/>
+            let mut skip_to_step_15 = false;
+
+            // These values missed their default value if we are not processing further.
+            let mut intersection_ratio = 0.;
+            let mut is_visible = false;
+
+            // Step 5
+            // > If the intersection root is not the implicit root, and target is not in
+            // > the same document as the intersection root, skip to step 11.
+            if !self.root_is_implicit_root() && *target.owner_document() != *document {
+                skip_to_step_15 = true;
+            }
+
+            // Step 6
+            // > If the intersection root is an Element, and target is not a descendant of
+            // > the intersection root in the containing block chain, skip to step 11.
+            // TODO(stevennovaryo): implement LayoutThread query that support this.
+            if let Some(_element) = self.maybe_element_root() {
+                debug!("descendant of containing block chain is not implemented");
+            }
+
+            // NOTE: Firefox skips these steps too, if target_rect or root_bounds is zero.
```

Comment:
Following Gecko's behavior and skipping these steps if ... (writing from the point of what we're doing here, not what FF is doing).

Reply:
I added it in https://github.com/servo/servo/pull/35551/commits/ecc67439f2459f2ef70304750c5abac9f41b2c91, I guess it is better to add it here since it does not make sense to assume null return from `getBoundingBox` as a zero rectangle.  

My bad it is more like an todo rather than a note as these is not implemented. It should also mention that it was for null value instead of zero (rectangle?).